docs/parts.md

```diff
@@ -0,0 +1,100 @@
+
+# Parts definition
+
+When is necessary to expose external components in the simulator that work together with the board (leds, sensors, speaker, headphone, etc.), it can be done by specifying the JSON definition of the part in the *pxtparts.ts* file using the following arguments:
+
+* **simulationBehavior** *(optional)*: built-in simulator logic name
+* **numberOfPins**: total number of pins including power, GPIO and other pins
+* **visual**: visual description or built-in visual name
+    1. **image** (optional):  URL to image asset
+    2. **builtIn** (optional): or name of built-in part visual
+    3. **width, height**: description of a parts visual; units don't matter but must be internally consistent
+    4. **pinDistance**: the distance between the centers of two adjecent pins; used to scale part for breadboard
+    5. **pinLocations**: the exact centers of each pin; must have as many locations as the "numberOfPins" property
+* **pinDefinitions**: metadata for each pin
+    1. **target**: e.g.: "ground", "MISO", "threeVolt", etc.
+    2. **style**: e.g.: "male", "female", "solder". 
+    3. **orientation**: e.g.: "+X", "-Z", etc.
```

Comment:
I believe "-Z" means "into the breadboard". TBH, I probably over engineered this field a bit. I don't really see many parts needing to use all 6 orientations. And I believe the allocator only understands two groups of orientations: into the breadboard (-Z) and everything else.